site/src/pages/RoadmapPage/YearModal.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { FC, useState } from 'react';
+import { Button, Form, Modal } from 'react-bootstrap';
+import { PlannerYearData } from '../../types/types';
+import './YearModal.scss';
+
+interface YearPopupQuarter {
+  id: string;
+  name: string;
+  checked?: boolean;
+}
+
+interface YearModalProps {
+  placeholderName: string;
+  placeholderYear: number;
+  show: boolean;
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  type: 'add' | 'edit';
+  saveHandler: (x: PlannerYearData) => void;
+  currentQuarters: string[];
+}
+
+const quarterValues: (selectedNames: string[]) => YearPopupQuarter[] = (data: string[]) => {
+  const base: YearPopupQuarter[] = [
+    { id: 'fall', name: 'Fall' },
+    { id: 'winter', name: 'Winter' },
+    { id: 'spring', name: 'Spring' },
+    { id: 'summer 1', name: 'Summer 1' },
+    { id: 'summer 2', name: 'Summer 2' },
+    { id: 'summer 10 Week', name: 'Summer 10 Week' },
+  ];
+  data.forEach((name) => {
+    const match = base.find((q) => q.id === name);
+    if (match) match.checked = true;
+  });
+  return base;
+};
```

Comment:
Being a little pedantic here but this function doesn't read as clearly as I like. I would change some variable names, id and name seem to be used interchangeably. Also, we could probably use null assertion here. Reads a little cleaner and there should never be a case where we select a nonexistent quarter ID, and if there is, we made a mistake and would probably benefit from the error that arises.

```suggestion
const quarterValues: (selectedQuarters: string[]) => YearPopupQuarter[] = (quarterIds: string[]) => {
  const base: YearPopupQuarter[] = [
    { id: 'fall', name: 'Fall' },
    { id: 'winter', name: 'Winter' },
    { id: 'spring', name: 'Spring' },
    { id: 'summer 1', name: 'Summer 1' },
    { id: 'summer 2', name: 'Summer 2' },
    { id: 'summer 10 Week', name: 'Summer 10 Week' },
  ];
  quarterIds.forEach((id) => {
    const quarter = base.find((q) => q.id === id)!;
    quarter.checked = true;
  });
  return base;
};
```

Reply:
the `id` is supposed to be consistent with other places in PPC, which have it written as `summer 1` and `summer 2` last I checked. The name can be changed though

Reply:
wait never mind you just showed otherwise in the DB.